types/tools/block-tool.d.ts

```diff
@@ -85,7 +85,7 @@ export interface BlockTool extends BaseTool {
  */
 export interface BlockToolConstructorOptions<D extends object = any, C extends object = any> {
   api: API;
-  data: BlockToolData<D>;
+  data?: BlockToolData<D>;
```

Comment:
By marking to undefinable, resolved https://github.com/codex-team/editor.js/pull/1480.

Reply:
It seems better to modify InlineToolConstructorOptions.
If you change BlockToolConstructorOptions.data to be optional, the user of this library asks "You might not pass data?" And can get confused.

Reply:
Like the PR I requested, InlineToolConstructorOptions.data can be added as an optional undefined type. "Why did this library have a type specified here?" As users would think, there is no need to significantly change the API design of this library, and as a result, I think there will be nothing wrong.

Reply:
👍
I think the solution in https://github.com/codex-team/editor.js/pull/1480 is better! 
In my solution, each BlockTool needs to check `undefined`. 

@neSpecc 
I changed this PR draft, and I want to merge https://github.com/codex-team/editor.js/pull/1480 first.

Reply:
Thank you for understanding!